docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+
+- The number of columns, `columns`, and the spacing between items, `spacing`, can be passed into the component.
+- Responsive number of columns (breakpoints)
+- Responsive spacing (breakpoints)
+
+**Nice-to-have**:
+
+- Server side rendering
+- Span multiple columns in a masonry
+- Performance
+- Logical tab order
+- Support for lazy loading/virtualisation
+
+In short, we wanted to build a masonry component that is light-weight enough to be performant and easy-to-use, but solid enough to do its job.
+
+## Two possible solutions: CSS Grid or CSS Flexbox?
+
+Since our goal was to develop a light-weight masonry component, an absolute positioning implementation, which most existing masonry libraries make use of, was not an option. It requires too much JavaScript, and therefore it is likely to be heavy-weight.
+
+This left either CSS Grid or CSS Flexbox, but each came with its own advantages and disadvantages.
+
+**1. CSS Grid + Configuring `grid-row-start` or `grid-row-end`of items**
+
+The idea is to create a grid container, `<Masonry />`, that contains grid items, `<MasonryItem />`s, each of which contains a child. Then, we compute the height of each child and set the height of its wrapper, `<MasonryItem />`, to this value. This allows every `<MasonryItem />` to be able to have a different height and to be large enough to display its child.
+
+One important thing to note is that simply setting a fixed height to `<MasonryItem />` will not do anything. Because it is a CSS grid item, it is constrained to the row height configured in its grid container, `<Masonry />`. So, we should use a property called `grid-row-start` or `grid-row-end`. Both properties span the height of a grid item to a specific number of rows. For example, applying `grid-row-end: span 10` to a `<MasonryItem />` spans its height to amount to 10 rows.
+
+This means that a `<MasonryItem />`'s height is always a multiple of the row height configured in `<Masonry />`. If the row height is `10px`, the height of `<MasonryItem />` is a multiple of `10px`: `100px`, `110px`, ... and so on. This limitation gives rise to a serious problem. However, there is always a solution, and both the problem and the solution are introduced in the next section.
+
+On the other hand, there is an amazing benefit to this implementation: it places a grid item to the shortest column by default. (Spoiler: this isn't the case in the other option). Also, this implementation makes it extremely handy to support the column spanning feature. We can use `grid-column-start` or `grid-column-end` to span the width of a grid item. Both of these advantages lead to less dependence on JavaScript, which meets our goal of making a light-weight component.
+
+Overall, this approach satisfies all the requirements discussed earlier except for server-side rendering. Still, this implementation seems like a fairly sufficient candidate to me. What do you think? Let's have a look at the other option, and it might be easier to decide.
```

Comment:
```suggestion
Overall, this approach satisfies all the requirements discussed earlier except for server-side rendering. Still, this implementation seemed like a fairly sufficient candidate to me. Let's have a look at the other option, and it might be easier to conclude how we came to the final implementation.
```

I would leave all questions and asking feedback/opinions in the end. Otherwise, it looks here like we are still developing the component